modules/api/pkg/handler/v2/provider/gcp.go

```diff
@@ -60,6 +63,61 @@ type gcpSubnetworksNoCredentialReq struct {
 	Network string
 }
 
+type GCPCommonReq struct {
+	ServiceAccount string
+	Credential     string
+}
+
+type GCPProjectCommonReq struct {
+	Request   GCPCommonReq
```

Comment:
This isn't exactly how the `Req` structs should look (to be fair it could, but we should be consistent with how things are implemented so far). What you want is to embed the common structures into your structs, so `GCPProjectCommonReq` should look like this:

```go
type GCPProjectCommonReq struct {
   common.ProjectReq // <- embed a project-aware req
   GCPCommonReq // <- embed the common req for GCP
   
   ProjectID string // <- the special field for this request
}
```

Respectively applies to all other structs as well, so `GCPProjectDatacenterReq` could look like this:

```go
type GCPProjectDatacenterReq struct {
	GCPProjectCommonReq
	DC             string
}
```

or

```go
type GCPProjectDatacenterReq struct {
    common.ProjectReq
	GCPCommonReq
	
	DC             string
}
```